rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/Transaction.java

```diff
@@ -374,6 +374,10 @@ public boolean isRecovering() {
                 (status.equals(LRAStatus.Cancelling) || status.equals(LRAStatus.Closing));
     }
 
+    public boolean isFailed() {
+        return status != null &&
```

Comment:
Thsis is not a request for a change but just a note "thinking loudly"
Do you know why the LRA `Transaction` instance permits the `status` to be `null`?
I think it could be better to assert that's not null? I can see the code may assign the `status` to `null` during recovery. Maybe some future change could adjust it.

Reply:
The status cannot be null. We used to use null to indicate that the transaction was active but we have since introduced an explicit state for this (`LRAStatus.Active`). I didn't mention it in my review since it is unrelated to the PR, I was going to clean up the code in one go in a dedicated PR.

Reply:
thanks Mike, that will be better.

Reply:
Where does "the code assign null to the status"? It should not do that.

Checking for status == null is a throwback to when we did not have an `LRAStatus.Active` state so the null checks are superfluous but I would advice against refactoring in the PR - do it in a separate one that does a general clean up of the code.

Reply:
@mmusgrov https://github.com/jbosstm/narayana/blob/master/rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/Transaction.java#L272